UnitTests/GitUI.Tests/CommitInfo/CommitInfoTests.cs

```diff
@@ -148,90 +133,132 @@ void SortAndCheckListsForEquality()
-            refs.Should().BeEquivalentTo(expected);
+                var expected = new Dictionary<string, int>
+                {
+                    ["refs/remotes/origin/master"] = 0,
+                    ["refs/heads/master"] = 1,
+                    ["refs/tags/v3.1"] = 2,
+                    ["refs/tags/v3.1 "] = 3,
+                    [" refs/tags/v3.1"] = 4
+                };
+
+                var refs = commitInfo.GetTestAccessor().GetSortedTags();
+
+                refs.Count.Should().Be(5);
+                refs.Should().BeEquivalentTo(expected);
+            });
         }
 
         [Test]
         public void GetSortedTags_should_remove_duplicate_refs()
         {
-            _gitExecutable.StageOutput("for-each-ref --sort=-taggerdate --format=\"%(refname)\" refs/tags/",
-                "refs/remotes/origin/master\nrefs/remotes/foo/duplicate\nrefs/remotes/foo/bar\nrefs/remotes/foo/duplicate\nrefs/remotes/foo/last"); // exact duplicates
-
-            var expected = new Dictionary<string, int>
+            RunCommitInfoTest(commitInfo =>
             {
-                ["refs/remotes/origin/master"] = 0,
-                ["refs/remotes/foo/duplicate"] = 1,
-                ["refs/remotes/foo/bar"] = 2,
-                ["refs/remotes/foo/last"] = 3,
-            };
+                _gitExecutable.StageOutput("for-each-ref --sort=-taggerdate --format=\"%(refname)\" refs/tags/",
+                    "refs/remotes/origin/master\nrefs/remotes/foo/duplicate\nrefs/remotes/foo/bar\nrefs/remotes/foo/duplicate\nrefs/remotes/foo/last"); // exact duplicates
+
+                var expected = new Dictionary<string, int>
+                {
+                    ["refs/remotes/origin/master"] = 0,
+                    ["refs/remotes/foo/duplicate"] = 1,
+                    ["refs/remotes/foo/bar"] = 2,
+                    ["refs/remotes/foo/last"] = 3,
+                };
+
+                var refs = commitInfo.GetTestAccessor().GetSortedTags();
 
-            var refs = _commitInfo.GetTestAccessor().GetSortedTags();
+                refs.Count.Should().Be(4);
+                refs.Should().BeEquivalentTo(expected);
+            });
+        }
+
+        private void RunCommitInfoTest(Action<GitUI.CommitInfo.CommitInfo> runTest)
+        {
+            UITest.RunControl(
+                createControl: form =>
+                {
+
+                    var uiCommandsSource = Substitute.For<IGitUICommandsSource>();
+                    uiCommandsSource.UICommands.Returns(x => _commands);
+
+                    // the following assignment of CommitInfo.UICommandsSource will already call this command
+                    _gitExecutable.StageOutput("for-each-ref --sort=-taggerdate --format=\"%(refname)\" refs/tags/", "");
+
+                    return new GitUI.CommitInfo.CommitInfo
+                    {
+                        Parent = form,
+                        UICommandsSource = uiCommandsSource
+                    };
```

Comment:
This common setup mock-up `_gitExecutable` must be used by the `CommitInfo` ctor before the tests modify the mock-up.